infrastructure/src/main/java/postech/soat/tech/challenge/api/controller/ProductController.java

```diff
@@ -0,0 +1,39 @@
+package postech.soat.tech.challenge.api.controller;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.*;
+import postech.soat.tech.challenge.api.request.ProductDTO;
+import postech.soat.tech.challenge.api.request.ProductRequestMapper;
+import postech.soat.tech.challenge.api.response.ApiResponse;
+import postech.soat.tech.challenge.model.Product;
+import postech.soat.tech.challenge.port.input.CreateProductUseCase;
+import postech.soat.tech.challenge.port.input.FindProductUseCase;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("/api/produto")
```

Comment:
O que acha de manter o identificador dos resources no plural? É uma boa prática de design de APIs pois torna os verbos semânticamente mais precisos, afinal, estamos sempre manipulando uma coleção de recursos.

Por exemplo, para consultar a lista de produtos, faria sentido que o GET seja sem nenhum parâmetro, no endpoint `/api/produtos`.

Da mesma forma, ao buscar um produto específico, faz sentido pensar que estamos especificando um pedido dentre um conjuto: `api/pedidos/123`.

```suggestion
@RequestMapping("/api/produtos")
```

Reply:
Boa.. mal bad..